libs/portals/admin/ids-admin/src/components/Client/Permissions.tsx

```diff
@@ -1,86 +1,110 @@
+import {
+  ClientFormTypes,
+  EditApplicationResult,
+  schema,
+} from '../forms/EditApplication/EditApplication.action'
 import { ShadowBox } from '../ShadowBox/ShadowBox'
-
-type Permission = {
-  id: string
-  label: string
-  description: string
-  api: string
-  locked?: boolean
-}
+import { useActionData, useParams } from 'react-router-dom'
+import { ClientContext } from '../../shared/context/ClientContext'
+import { AuthAdminClientAllowedScope } from '@island.is/api/schema'
+import AddPermissions from '../forms/AddPermissions/AddPermissions'
+import { getTranslatedValue } from '@island.is/portals/core'
+import isEqual from 'lodash/isEqual'
 
 interface PermissionsProps {
-  data?: Permission[]
+  allowedScopes?: AuthAdminClientAllowedScope[]
 }
 
-const mockData: NonNullable<PermissionsProps['data']> = [
-  {
-    label: 'Staða og hreyfingar',
-    id: '@island.is/finance:overview',
-    description:
-      'Skoða stöðu við ríkissjóð og stofnanir, hreyfingar, greiðsluseðla og greiðslukvittanir.',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Full Access',
-    id: '@island.is/auth/admin:full',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Skattskýrslur',
-    id: '@skatturinn.is/skattskyrslur',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Skatturinn',
-    locked: true,
-  },
-  {
-    label: 'Staða og hreyfingar',
-    id: '@island.is/finance',
-    description:
-      'Skoða stöðu við ríkissjóð og stofnanir, hreyfingar, greiðsluseðla og greiðslukvittanir.',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Full Access',
-    id: '@island.is/auth/admin',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-]
-
-function Permissions({ data = mockData }: PermissionsProps) {
-  const { formatMessage } = useLocale()
+function Permissions({ allowedScopes }: PermissionsProps) {
+  const { formatMessage, locale } = useLocale()
   const [isModalVisible, setIsModalVisible] = useState(false)
+  const params = useParams()
+  const [permissions, setPermissions] = useState<AuthAdminClientAllowedScope[]>(
+    allowedScopes ?? [],
+  )
+  const actionData = useActionData() as EditApplicationResult<
+    typeof schema[typeof ClientFormTypes.permissions]
+  >
+
+  const actionDataRef = useRef(actionData?.data)
+
+  useEffect(() => {
+    if (actionData?.intent === ClientFormTypes.permissions) {
+      if (!isEqual(actionData?.data, actionDataRef?.current)) {
+        if (actionData?.data) {
+          setAddedScopes([])
+          setRemovedScopes([])
+        }
+        if (actionData?.globalError) {
+          toast.error(formatMessage(m.globalErrorMessage))
+        }
+      }
+    }
```

Comment:
If there is `globalError` we don't have data so this if statements can be simplified.
```suggestion
    if (actionData?.intent === ClientFormTypes.permissions) {
        if (actionData?.globalError) {
          toast.error(formatMessage(m.globalErrorMessage))
        }
       else if (actionData?.data) {
          setAddedScopes([])
          setRemovedScopes([])
        }
    }
```